File: 33 - Magic 8 Ball/C++/main.cpp
#include <iostream>
#include <random>

// returns a random integer in the range passed
int random_int(int low, int high)
{
    std::random_device                 rand_dev;
    std::mt19937                       generator(rand_dev());
    std::uniform_int_distribution<int> distr(low, high);
    return distr(generator);
}

int main()
{
    std::string RESPONSES[] = {
        "Yes",
        "No",
        "Maybe",
        "Ask again later",
    };
    std::cout << "What is your question? ";
    std::string _;
    std::cin >> _;
    std::cout << RESPONSES[random_int(0, RESPONSES->size())] << std::endl;
    return 0;
}

File: 33 - Magic 8 Ball/Golang/main.go
// Magic 8 ball

package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"time"
)

var RESPONSES = []string{
	"No",
	"Yes",
	"Maybe",
	"Ask again later",
}

func randomInt(low, high int) int {
    rand.Seed(time.Now().UnixNano())
	return rand.Intn(high-low) + low
}

func main() {
	fmt.Print("What is your question? ")
	reader := bufio.NewReader(os.Stdin)
    reader.ReadString('\n') // no variables taken :p
	randomNumber := randomInt(0, len(RESPONSES))
	fmt.Println(RESPONSES[randomNumber])
}
